cmds/upspin_sos/service.go

```diff
@@ -4,32 +4,128 @@
 
 package main
 
-type DummyUpspinService struct {
+import (
+	"flag"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"regexp"
+	"strings"
+)
+
+var (
+	upspinConfigDir = flag.String("configdir", filepath.Join(os.Getenv("HOME"), "upspin"), "path for Upspin config file")
+	upspinKeyDir    = flag.String("keydir", filepath.Join(os.Getenv("HOME"), ".ssh"), "path for username directory to hold key files")
+)
+
+type UpspinService struct {
 	Configured bool
 	User       string
 	Dir        string
 	Store      string
 	Seed       string
 }
 
-func (us *DummyUpspinService) ToggleFlag() {
+func makeUserDirectories(dir string) error {
+	if _, err := os.Stat(dir); os.IsNotExist(err) {
+		if err := os.MkdirAll(dir, 0777); err != nil {
+			return err
+		}
+		return filepath.Walk(dir, func(name string, info os.FileInfo, err error) error {
+			if err == nil {
+				err = os.Chown(name, 1000, 1000)
+			}
+			return err
+		})
+	}
+	return nil
+}
+
+func getFileData(path string) map[string]string {
```

Comment:
why pass in the directory? why not pass in the whole filename instead of hardcoding "config" in this function?